guides/source/upgrading_ruby_on_rails.md

```diff
@@ -169,93 +169,109 @@ more meaningful preview images. Previously the first frame of the video would be
 ```
 
-If you forget to do this, when you switch to vips you will see this error: `unable to call crop: you supplied 2 arguments, but operation needs 5`.
+If you don't do this, when you switch to vips you will see this error: `unable to call crop: you supplied 2 arguments, but operation needs 5`.
 
 #### Clamp your crop values:
+
 Vips is more strict than ImageMagick when it comes to cropping:
+
 1. It will not crop if `x` and/or `y` are negative values. e.g.: `[-10, -10, 100, 100]`
 2. It will not crop if position (`x` or `y`) plus crop dimension (`width`, `height`) is larger than the image. e.g.: a 125x125 image and a crop of `[50, 50, 100, 100]`
 
-If you forget to do this, when you switch to vips you will see this error: `extract_area: bad extract area`
+If you don't do this, when you switch to vips you will see this error: `extract_area: bad extract area`
 
-#### Adjust the background color used in resize and pad
+#### Adjust the background color used for `resize_and_pad`
 Vips uses black as the default background color `resize_and_pad`, instead of white like ImageMagick. Fix that by using the `background` option:
+
 ```diff
 - variant(resize_and_pad: [300, 300])
 + variant(resize_and_pad: [300, 300, background: [255]])
 ```
 
 #### Remove any EXIF based rotation
 Vips will auto rotate images using the EXIF value when processing variants. If you were storing rotation values from user uploaded photos to apply rotation with ImageMagick, you must stop doing that:
+
 ```diff
 - variant(format: :jpg, rotate: rotation_value)
 + variant(format: :jpg)
 ```
 
 #### Replace monochrome with colourspace
 Vips uses a different option to make monochrome images:
+
 ```diff
 - variant(monochrome: true)
 + variant(colourspace: "b-w")
 ```
 
 #### Switch to libvips options for compressing images
 JPEG
+
 ```diff
 - variant(strip: true, quality: 80, interlace: "JPEG", sampling_factor: "4:2:0", colorspace: "sRGB")
 + variant(saver: { strip: true, quality: 80, interlace: true })
 ```
 
 PNG
+
 ```diff
 - variant(strip: true, quality: 75)
 + variant(saver: { strip: true, compression: 9 })
 ```
 
 WEBP
+
 ```diff
 - variant(strip: true, quality: 75, define: { webp: { lossless: false, alpha_quality: 85, thread_level: 1 } })
 + variant(saver: { strip: true, quality: 75, lossless: false, alpha_q: 85, reduction_effort: 6, smart_subsample: true })
 ```
 
 GIF
+
 ```diff
 - variant(layers: "Optimize")
 + variant(saver: { optimize_gif_frames: true, optimize_gif_transparency: true })
 ```
 
 #### Deploy to production
 Active Storage encodes into the url for the image the list of transformations that must be performed.
-If you app is caching these urls, your images will break after you deploy the new code to production.
+If your app is caching these urls, your images will break after you deploy the new code to production.
 Because of this you must manually invalidate your affected cache keys.
 
 For example, if you have something like this in a view:
-```rhtml
+
+```ruby
```

Comment:
I think this should stay `rhtml`, it just requires the new line above it.
The following example uses `ruby` for the first example and `rhtml` for the second one:
<img width="678" alt="image" src="https://user-images.githubusercontent.com/28561/129712699-c9d1c4b8-0716-4dd5-90f8-789a936619d4.png">